util.py

```diff
@@ -39,39 +43,55 @@ def toAlpha3Code(code):
         return iso639CodesInverse[code] if code in iso639CodesInverse else code
 
 
-def getLocalizedLanguages(locale, dbPath, languages=[]):
+def getLanguageNames(locale, dbPath):
     global langNamesDBConn
+
     if not langNamesDBConn:
         if os.path.exists(dbPath):
             langNameDBConn = sqlite3.connect(dbPath)
-            c = langNameDBConn.cursor()
+            cursor = langNameDBConn.cursor()
         else:
             logging.error('Failed to locate language name DB: %s' % dbPath)
-            return {}
+            return None
+
+    return cursor.execute('SELECT * FROM languageNames WHERE lg=?', (locale, )).fetchall()
```

Comment:
Could this ever result in multiple concurrent threads accessing the same cursor? The cursor isn't thread-safe.

Reply:
I'm guessing also multiple processes (starting servlet with -j2) would mess this up.

Reply:
Okay, then let's make `ThreadPoolExecutor` a global and keep it single-threaded, so that we can guarantee only one access at a time.